Makefile

```diff
@@ -228,8 +230,15 @@ test-integration: clean-cov generate fmt vet ginkgo ## Run Integration tests.
 		--coverpkg $(INTEGRATION_COVER_PKGS) \
 		--output-dir $(PROJECT_PATH)/coverage/integration \
 		--coverprofile cover.out \
-		--fail-fast \
 		-v \
+		--compilers=$(INTEGRATION_TEST_NUM_CORES) \
+		--procs=$(INTEGRATION_TEST_NUM_PROCESSES) \
```

Comment:
what about `-p` ??

```
 -p 
    If set, ginkgo will run in parallel with an auto-detected number of nodes.

```

Reply:
I can give `-p` a try, caveat is that https://onsi.github.io/ginkgo/#recommended-continuous-integration-configuration does not recommend using it in CI, although not use it is an issue with using it via github actions CI 🤔 
```
-procs=N will run each suite in parallel. This can substantially speed up suites and you should experiment with different values of N. Note that it is not recommended that you run specs in parallel with -p on CI. Some CI services run on shared machines that will report (e.g.) 32 cores but will not actually give an individual account access to all those compute resources!
```

Reply:
I see the issue with CI envs... ok. Let's give it a try

Reply:
Tried to use `-p` in https://github.com/Kuadrant/limitador-operator/pull/134/commits/5a59106568bebef372b8a6d3a7d1be6170c3fe5b and looks like it's using 4 processes for the tests - https://github.com/Kuadrant/limitador-operator/actions/runs/9081747820/job/24956406772?pr=134 which matches the number specified at https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for-public-repositories

![image](https://github.com/Kuadrant/limitador-operator/assets/24636860/7bd7e9f3-549b-412e-baef-28d9bfd7f317)

It's about 1.5 mins slower than previous where we able to specify to run using 10 processes instead 🤔 

Probably will revert to the previous commit since it looks like github CI is able to support it unless you want to go with `-p` instead